crates/locksmith/src/error.rs

```diff
@@ -0,0 +1,27 @@
+#[derive(Debug)]
+pub enum LocksmithErrorKind {
+    LocksmithTimeout,
+    LocksmithPoisonError,
+    LocksmithWouldBlock,
+}
+
+#[derive(Debug)]
+pub enum LockType {
+    Lock,
+    Read,
+    Write,
+}
+
+#[derive(Debug)]
+pub struct LocksmithError {
+    lock_type: LockType,
+    kind: LocksmithErrorKind,
+}
```

Comment:
just for rust idiom, even if you don't override any of the provided functions:

```suggestion
}

impl std::error::Error for LocksmithError {
}
```